### DISCUSSION TITLE
Where can I find "configuration options" for the (query) client?

### DISCUSSION THREAD

**Discussion by vfonic** [TanStack/query#5834] [Q&A]
I'm using React and here's how I initialize `QueryClientProvider` in my `App.tsx`:

```ts
const queryClient = new QueryClient()

const App = () => (
  <QueryClientProvider client={queryClient}>
    <Whatever />
  </QueryClientProvider>
)
export default App
```

I found out that default `QueryClient` configuration is a bit too aggressive for my liking and I wanted to change it.

I went to the documentation site (I use v4, which is latest).
Checked here: https://tanstack.com/query/v4/docs/react/guides/important-defaults
And here: https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientgetdefaultoptions

Searched for: `default`, `option`...

I found some scarce options documentation in the examples and here:
https://tanstack.com/query/v4/docs/react/reference/useQuery

I think `new QueryClient()` config options documentation is extremely important. Am I missing something obvious here?

**Comment by vfonic** [comment]
Even TypeScript didn't get me too far:

```ts
export interface DefaultOptions<TError = unknown> {
  queries?: QueryObserverOptions<unknown, TError>
  mutations?: MutationObserverOptions<unknown, TError, unknown, unknown>
}
```

**Comment by TkDodo** [comment]
it's basically:

```
new QueryClient({
  defaultOptions: {
    queries: {}
  }
})
```

and then inside Queries, you put in whatever you would normally pass to `useQuery` (that's  the `QueryObserverOptions` you found).

An example can also be found here: https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclient

**Comment by vfonic** [comment]
Yes! That's the page I was looking for and couldn't find.

I suggest renaming this page or make it more obvious/findable.

...and it's misleading to say

> The `QueryClient` can be used to interact with a cache

Yes, it's true, it _can_ be used to interact with the cache, but it's actually much much more than that:

1. It's _required_ to be used to interact with cache
2. It's required to be configured/instantiated in order to make any queries and/or queries to cache
3. It's the most integral part of the library. Without `QueryClient`, you can't do pretty-much anything (correct me if I'm wrong here)
